Kernel/Devices/HID/HIDManagement.cpp

```diff
@@ -111,9 +111,6 @@ UNMAP_AFTER_INIT ErrorOr<void> HIDManagement::enumerate()
         return {};
     m_i8042_controller = I8042Controller::initialize();
 
-    // Note: If we happen to not have i8042 just return "gracefully" for now.
-    if (!m_i8042_controller->check_existence({}))
```

Comment:
Please don't remove this check. It seems like Linux does this check too and doesn't rely on ACPI.
I wrote it on the discord server:

> supercomputer7: ACPI on QEMU doesn't indicate if i8042 is present
[8:26 AM] supercomputer7: not in the FADT
[8:26 AM] supercomputer7: they encode it in the AML
[8:26 AM] supercomputer7: as for the check existence, it is fine to do that early
[8:26 AM] supercomputer7: Linux does that too
[8:27 AM] supercomputer7: osdev wiki seems to be wrong about this

Reply:
Instead of doing the self-test, we can do what Linux does on init (boot):
We can "flush" the registers of the controller to see if it exists or not.

Linux does a self-test on resume apparently, on boot it doesn't do that. Flushing the registers is quite similar to what we do in `drain_output_buffer` method of the `I8042Controller`. Look at this to see what I talk about:
https://github.com/torvalds/linux/blob/4f12b742eb2b3a850ac8be7dc4ed52976fc6cb0b/drivers/input/serio/i8042.c#L264-L292

Reply:
Can we instead go with an OR approach? If the FADT says it's there, just assume it is, and only if it doesn't do this new flush check you're suggesting?

Reply:
Probing check restored, using Idan's suggested approach now.